client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -74,10 +74,11 @@ private final class PoolManager[F[_], A <: Connection[F]](
   private def numConnectionsCheckHolds(key: RequestKey): Boolean =
     curTotal < maxTotal && allocated.getOrElse(key, 0) < maxConnectionsPerRequestKey(key)
 
-  private def isExpired(k: RequestKey, t: Instant): Boolean =
-    waitExpiryTime(k).isFinite() && t
-      .plus(waitExpiryTime(k).toSeconds, ChronoUnit.SECONDS)
-      .isBefore(Instant.now())
+  private def isExpired(t: Instant): Boolean = {
+    val elapsed = Instant.now().getEpochSecond - t.getEpochSecond
+    (requestTimeout.isFinite() && elapsed >= requestTimeout.toSeconds) || (responseHeaderTimeout
```

Comment:
I am feeling stupid.  What's the difference between requestTimeout and responseHeaderTimeout?  They both seem to be enforced in the same place.

Reply:
I am not sure what you mean. Any of them could have timed out, we need to check for both, no?

Reply:
I don't see any difference in how these two properties are enforced in this function, or anywhere else.  What is the semantic difference between them?

Reply:
I may have the wrong interpretation of the two. What happens when requestTimeout is 2 seconds and responseHeaderTimeout is 1 second? responseHeaderTimeout can timeout independent of requestTimeout and vice versa. Is this correct?

Reply:
I don't understand how they're being enforced such that they're two different properties.  They're both being compared to the same elapsed time.

Reply:
responseHeaderTimeout not set
requestTimeout set

if we check only for responseHeaderTimeout we miss requestTimeout

responseHeaderTimeout set to 1 sec
requestTimeout set to 2 sec

if we only check for requestTimeout and we start after 1 sec and finish before 2 sec we are ok  but we missed responseHeaderTimeout

etc. What am I missing?

Reply:
There are two properties on the config.  Both of them are enforced right here, based on the same elapsed time.  i.e., the clock on one does not start ticking before the other.  Effectively, we're enforcing a single timeout that is `min(requestTimeout, responseHeaderTimeout)`, no?

Reply:
Yes.

Reply:
So if the property we're enforcing is just a min of two properties, why two properties?  I would think that the start or end time we're measuring against would be different.